folium/plugins/time_slider_choropleth.py

```diff
@@ -26,21 +26,25 @@ class TimeSliderChoropleth(JSCSSMixin, Layer):
         Whether the Layer will be included in LayerControls.
     show: bool, default True
         Whether the layer will be shown on opening (only for overlays).
-
+    init_timestamp: int, default 0
+        Initial time-stamp index on the slider. Must be in the range
+        `[-L, L-1]`, where `L` is the maximum number of time stamps in
+        `styledict`. For example, use `-1` to initialize the slider to the
+        latest timestamp.
     """
     _template = Template(u"""
         {% macro script(this, kwargs) %}
             var timestamps = {{ this.timestamps|tojson }};
             var styledict = {{ this.styledict|tojson }};
-            var current_timestamp = timestamps[0];
-
+            var current_timestamp = timestamps[
+                (timestamps.length+{{ this.init_timestamp}}) % timestamps.length];
             // insert time slider
             d3.select("body").insert("p", ":first-child").append("input")
                 .attr("type", "range")
                 .attr("width", "100px")
                 .attr("min", 0)
                 .attr("max", timestamps.length - 1)
-                .attr("value", 0)
+                .attr("value", current_timestamp)
```

Comment:
The `"value"` here is the index if I'm not mistaken

```suggestion
                .attr("value", {{ this.init_timestamp }})
```